pkg/supportbundle/spec.go

```diff
@@ -106,106 +106,200 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
 			}
-
-			logger.Debugf("created %q default support bundle spec secret", app.GetSlug())
-			return supportBundle, nil
-		} else {
-			return nil, errors.Wrap(err, "failed to read support bundle secret")
 		}
 	}
 
-	if existingSecret.Data == nil {
-		existingSecret.Data = map[string][]byte{}
+	// Include discovered support bundle specs and default kotsadm support bundle spec. Perform this action here so
+	// as not to add discovered specs to the default support bundle spec secret.c
+	return addDiscoveredSpecs(builtBundles["kotsadm"], app, clientset), nil
+}
+
+// addClusterSpecificSpec adds cluster specific and upload results URI to the support bundle
+func addClusterSpecificSpec(app apptypes.AppType, b *troubleshootv1beta2.SupportBundle, opts types.TroubleshootOptions, namespacesToCollect []string, namespacesToAnalyze []string, imageName string, pullSecret *troubleshootv1beta2.ImagePullSecrets) *troubleshootv1beta2.SupportBundle {
+	supportBundle := b.DeepCopy()
+
+	// determine an upload URL
+	var uploadURL string
+	var redactURL string
+	randomBundleID := strings.ToLower(rand.String(32))
+	if opts.DisableUpload {
+		//Just use the library internally
+		return supportBundle
+	} else if opts.Origin != "" {
+		uploadURL = fmt.Sprintf("%s/api/v1/troubleshoot/%s/%s", opts.Origin, app.GetID(), randomBundleID)
+		redactURL = fmt.Sprintf("%s/api/v1/troubleshoot/supportbundle/%s/redactions", opts.Origin, randomBundleID)
+	} else if opts.InCluster {
+		uploadURL = fmt.Sprintf("%s/api/v1/troubleshoot/%s/%s", fmt.Sprintf("http://kotsadm.%s.svc.cluster.local:3000", util.PodNamespace), app.GetID(), randomBundleID)
+		redactURL = fmt.Sprintf("%s/api/v1/troubleshoot/supportbundle/%s/redactions", fmt.Sprintf("http://kotsadm.%s.svc.cluster.local:3000", util.PodNamespace), randomBundleID)
+	} else {
+		uploadURL = fmt.Sprintf("%s/api/v1/troubleshoot/%s/%s", os.Getenv("API_ADVERTISE_ENDPOINT"), app.GetID(), randomBundleID)
+		redactURL = fmt.Sprintf("%s/api/v1/troubleshoot/supportbundle/%s/redactions", os.Getenv("API_ADVERTISE_ENDPOINT"), randomBundleID)
 	}
-	existingSecret.Data[SpecDataKey] = renderedSpec
-	existingSecret.ObjectMeta.Labels = labels
 
-	_, err = clientset.CoreV1().Secrets(util.PodNamespace).Update(context.TODO(), existingSecret, metav1.UpdateOptions{})
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to update support bundle secret")
+	supportBundle.Spec.AfterCollection = []*troubleshootv1beta2.AfterCollection{
+		{
+			UploadResultsTo: &troubleshootv1beta2.ResultRequest{
+				URI:       uploadURL,
+				Method:    "PUT",
+				RedactURI: redactURL,
+			},
+		},
 	}
 
-	// Include discovered support bundle specs. Perform this action here so
-	// as not to add discovered specs to the default support bundle spec secret.
-	return addDiscoveredSpecs(supportBundle, app, clientset), nil
+	supportBundle = addDefaultDynamicTroubleshoot(supportBundle, app, imageName, pullSecret)
```

Comment:
This spec seems more like a `"dynamic-default"` rather than a `"cluster-specific"` since it's just the dynamic support bundle spec for this particular instance+app.

Reply:
when I was checking the content of dynamic default spec, I saw some collector had cluster specfic information, that's why I moved it into  cluster-specific spec. 
```
collectorName: app-info.json
      data: |-
        {
          "instance_id": "*****",
          "cluster_id": "******",
```

As your pointed out, it is for  kots instance+app, I think it is ok to move it back to default spec, does it sound good to you?

Reply:
i think it's fine to move it back into the default for now